command/resource/helper.go

```diff
@@ -233,3 +225,51 @@ func (resource *Resource) List(gvk *GVK, q *client.QueryOptions) (*ListResponse,
 
 	return out, nil
 }
+
+func inferGVKFromResourceType(resourceType string) (*GVK, error) {
+	s := strings.Split(resourceType, ".")
+	if len(s) == 1 {
+		kindToGVKMap := BuildKindToGVKMap()
+		kind := strings.ToLower(s[0])
+		if len(kindToGVKMap[kind]) == 0 {
+			return nil, fmt.Errorf("The shorthand name does not map to any existing resource type, please check `consul api-resources`")
+		} else if len(kindToGVKMap[kind]) == 1 {
+			// infer gvk from resource kind
+			gvkSplit := strings.Split(kindToGVKMap[kind][0], ".")
+			return &GVK{
+				Group:   gvkSplit[0],
+				Version: gvkSplit[1],
+				Kind:    gvkSplit[2],
+			}, nil
+		} else {
+			return nil, fmt.Errorf("The shorthand name has conflicts %v, please use the full name", kindToGVKMap[s[0]])
+		}
+	}
+
+	if len(s) != 3 {
+		return nil, fmt.Errorf("Must provide resource type argument with either in group.verion.kind format or its shorthand name")
+	}
+
+	return &GVK{
+		Group:   s[0],
+		Version: s[1],
+		Kind:    s[2],
+	}, nil
+}
```

Comment:
I think this is a great candidate to be converted to a switch statement

Reply:
Right! used switch to make the statements more readable